res/controllers/Numark-Mixtrack-Platinum-FX-scripts.js

```diff
@@ -0,0 +1,1899 @@
+// eslint-disable-next-line no-var
+var MixtrackPlatinumFX = {};
+
+// FX toggles
+MixtrackPlatinumFX.toggleFXControlEnable = true;
+MixtrackPlatinumFX.toggleFXControlSuper = false;
+
+MixtrackPlatinumFX.shifBrowseIsZoom = false;
+
+// setting this to false sets tap the file bpm, but without a way to reset its dangerous
+MixtrackPlatinumFX.tapChangesTempo = true;
+
+// pitch ranges
+// add/remove/modify steps to your liking
+// default step must be set in Mixxx settings
+// setting is stored per deck in pitchRange.currentRangeIdx
+MixtrackPlatinumFX.pitchRanges = [0.08, 0.16, 0.5];
+
+MixtrackPlatinumFX.HIGH_LIGHT = 0x7F;
+MixtrackPlatinumFX.LOW_LIGHT = 0x01;
+
+// whether the corresponding Mixxx option is enabled
+// (Settings -> Preferences -> Waveforms -> Synchronize zoom level across all waveforms)
+MixtrackPlatinumFX.waveformsSynced = true;
+
+// jogwheel
+MixtrackPlatinumFX.jogScratchSensitivity = 1024;
+MixtrackPlatinumFX.jogScratchAlpha = 1; // do NOT set to 2 or higher
+MixtrackPlatinumFX.jogScratchBeta = 1/32;
+MixtrackPlatinumFX.jogPitchSensitivity = 10;
+MixtrackPlatinumFX.jogSeekSensitivity = 10000;
+
+// blink settings
+MixtrackPlatinumFX.enableBlink = true;
+MixtrackPlatinumFX.blinkDelay = 700;
+
+// autoloop sizes, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatloop_X_toggle
+MixtrackPlatinumFX.autoLoopSizes = [
+    "0.0625",
+    "0.125",
+    "0.25",
+    "0.5",
+    "1",
+    "2",
+    "4",
+    "8"
+];
+
+// beatjump values, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatjump_X_forward
+// underscores (_) at the end are needed because numeric values (e.g. 8) have two underscores (e.g. beatjump_8_forward),
+// but "beatjump_forward"/"beatjump_backward" have only one underscore
+MixtrackPlatinumFX.beatJumpValues = [
+    "0.0625_",
+    "0.125_",
+    "0.25_",
+    "0.5_",
+    "1_",
+    "2_",
+    "", // "beatjump_forward"/"beatjump_backward" - jump by the value selected in Mixxx GUI (4 by default)
+    "8_"
+];
+
+// dim all lights when inactive instead of turning them off
+components.Button.prototype.off = MixtrackPlatinumFX.LOW_LIGHT;
+
+// pad modes control codes
+MixtrackPlatinumFX.PadModeControls = {
+    HOTCUE: 0x00,
+    AUTOLOOP: 0x0D,
+    FADERCUTS: 0x07,
+    SAMPLE1: 0x0B,
+    BEATJUMP: 0x01, // DUMMY not used by controller
```

Comment:
can you elaborate? what do you mean with not used by controller?

Reply:
Beatjump is working on the map. It might just need correction to 
BEATJUMP: 0x02